src/routers/handlers/accounts_submitted/accounts.submitted.ts

```diff
@@ -0,0 +1,68 @@
+import { PrefixedUrls } from "../../../utils/constants/urls";
+import { BaseViewData, GenericHandler } from "../generic";
+import { Request, Response } from "express";
+import { getCompanyName, getCompanyNumber, getTransactionId, getUserProfile } from "./../../../utils/session";
+import { CompanyProfile } from "@companieshouse/api-sdk-node/dist/services/company-profile";
+import { checkCompanyNumberFormatIsValidate } from "./../../../utils/format/company.number.format";
+
+interface AccountsSubmittedViewData extends BaseViewData {
+        transactionId: string | Error,
+        companyProfile: Pick<CompanyProfile, "companyName" | "companyNumber">
+        payment: string
+        userEmail: string
+}
+
+export class AccountsSubmittedHandler extends GenericHandler{
+    constructor() {
+        super({
+            title: "Accounts Submitted",
+            backURL: null
+        });
+    }
+
+    async execute(req: Request, _response: Response
+    ): Promise<AccountsSubmittedViewData> {
+
+        super.populateViewData(req);
+        this.baseViewData.backURL = PrefixedUrls.PAYMENT;
+
+        if (typeof req.session === "undefined"){
+            throw new Error("Session not properly set");
+        }
+
+        const companyName = getCompanyName(req.session);
+        const transactionId = getTransactionId(req.session);
+        const companyNumber = getCompanyNumber(req.session);
+        const userEmail = getUserProfile(req.session!)!.email;
+
+        // Payment will be set on the previous page
+        const payment = this.getPaymentFromQuery(req.query);
```

Comment:
You should use the package type in the session to work out if a payment is needed, and how much that is. Overseas and CIC need payment, the overs don't. I'm not a fan of using a query parameter for this.